tests/backwards_compatibility_tests/test_general_compatibility.py

```diff
@@ -0,0 +1,198 @@
+            'tags': ['light'],
+            'caption': 'example_doc_3'
+        },
+        {
+            "_id": f"example_doc_4",
+            "text_field": "Red Bus",
+            "image_field": "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image4.jpg",
+            'tags': ['bus', 'red'],
+            'caption': 'example_doc_4'
+        }
+    ]
+
+    extra_docs = [
+        {
+            '_id': 'example_doc_5',
+            'text_field': 'Woman looking at phone',
+            'image_field': 'https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image0.jpg',
+            'tags': ['women', 'phone'],
+            'my_int': 3
+        },
+        {
+            "_id": f"example_doc_6",
+            "text_field": "Woman skiing",
+            "image_field": "https://raw.githubusercontent.com/marqo-ai/marqo-clip-onnx/main/examples/coco.jpg",
+            'tags': ['ski'],
+            'caption': 'example_doc_6'
+        },
+    ]
+    indexes_to_test_on = [structured_index_metadata, unstructured_index_metadata]
+    queries = ["travel", "horse light", "travel with plane"]
+    search_methods = ["TENSOR", "LEXICAL", "HYBRID"]
+    result_keys = search_methods # Set the result keys to be the same as search methods for easy comparison
+    searchable_attributes = {"TENSOR": ['image_field', 'multimodal_field'], "LEXICAL": ['text_field']}
+
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.client = marqo.Client(**cls.client_settings)
+        cls.logger.debug(f"Creating indexes {cls.indexes_to_test_on}")
+        cls.create_indexes(cls.indexes_to_test_on)
+        cls.indexes_to_delete = [cls.structured_index_metadata['indexName'],
+                                 cls.unstructured_index_metadata['indexName']]
+
+    def prepare(self):
+        """
+        Prepare the indexes and add documents for the test.
+        Also store the search results for later comparison.
+        """
+        with self.subTest("Adding documents to structured and unstructured indexes"):
+            try:
+                self.logger.debug(f'Feeding documents to {self.indexes_to_test_on}')
+                for index in self.indexes_to_test_on:
+                    if index.get("type") is not None and index.get('type') == 'structured':
+                        self.client.index(index_name=index['indexName']).add_documents(documents=self.docs)
+                    else:
+                        self.client.index(index_name=index['indexName']).add_documents(documents=self.docs,
+                                                                                       mappings=self.mappings,
+                                                                                       tensor_fields=self.tensor_fields)
+            except Exception as e:
+                self.logger.error(f"Exception occurred while adding documents {e}")
```

Comment:
Should we raise e again so that the test run is aborted? If the prepare step is not properly run, we should not proceed to the test step, right?

Reply:
[changed everywhere](https://github.com/marqo-ai/marqo/pull/1024/commits/740097f76bc5fd67568d409c1a680d9ebcb77333#diff-8e63438eee12f7525b6324b87ee6d34a895fad3f38da711c9880c7475d1a565cR152)